Update ConfigDef to have more metadata and use it for validation

varunpuranik opened the pull request:
- Added more metadata to the connector ConfigDef for use with control center
- Extending from AbstractConfig to be able to use it for parsing and validation.
- Also marked Kafka dependencies as "provided" in sbt to avoid packaging them in the jar.

varunpuranik committed:
Merge branch 'master' into dev-5.0

varunpuranik committed:
Update ConfigDef to have more metadata and use it for validation

- Added more metadata to the connector ConfigDef for use with control
center
- Extending from AbstractConfig to be able to use it for parsing and
validation.

msftclas commented:
Hi __@varunpuranik__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Varun Puranik). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla.microsoft.com.
    </p>

TTYL, MSBOT;

dluc commented on build.sbt:
```diff
@@ -19,8 +19,8 @@ libraryDependencies ++= {
-    "org.apache.kafka" % "connect-json" % kafkaVersion,
+    "org.apache.kafka" % "connect-api" % kafkaVersion % "provided",
+    "org.apache.kafka" % "connect-json" % kafkaVersion % "provided",
     "com.microsoft.azure" % "azure-eventhubs" % azureEventHubSDKVersion,
```
does the provided package match the version required? I'm wondering what happens if we expect version "foo.X" and the provided is "foo.Y"

varunpuranik replied:
The connector expects Kafka v0.10 to be installed. I have added a line to the README to say so.

dluc commented on src/test/scala/com/microsoft/azure/iot/iothub2kafka/IotHubSourceTaskTest.scala:
```diff
@@ -11,6 +11,8 @@ import org.scalatest.{FlatSpec, GivenWhenThen}
 
 class IotHubSourceTaskTest extends FlatSpec with GivenWhenThen with JsonSerialization {
 
+  // This test is ignored as the configuration values in the test application.conf has no values.
```
nit: typo s/has/have/

dluc reviewed (approved):
Just a question about the behavior of "provided" dependencies

varunpuranik committed:
Fixing comments in README and test cases

- Addressed comments from Pull Request
- Fixed README to specify that Kafka Version 0.10 is required.
- Also fixed typo in a comment.
